Makefile

```diff
@@ -1,19 +1,34 @@
-# The following parameters are required:
+# The following parameters are required by 'run':
 # - REPOSITORY
 # - CIRCLECI_API_KEY
 #
 
-AWS_ACCESS_KEY_ID := $(shell aws secretsmanager get-secret-value --secret-id=ci-user-aws-keys${randomSeed} | jq -r '.SecretString'| jq -r .access_key_id)
-AWS_SECRET_ACCESS_KEY := $(shell aws secretsmanager get-secret-value --secret-id=ci-user-aws-keys${randomSeed} | jq -r '.SecretString'| jq -r .secret_key)
-GITHUB_ORG := $(shell echo ${REPOSITORY} | cut -d "/" -f 2)
-GITHUB_REPO := $(shell echo ${REPOSITORY} | cut -d "/" -f 3)
-
 run:
 	@echo "Set CIRCLECI environment variables\n"
+	AWS_ACCESS_KEY_ID := $(shell aws secretsmanager get-secret-value --secret-id=ci-user-aws-keys${randomSeed} | jq -r '.SecretString'| jq -r .access_key_id)
+	AWS_SECRET_ACCESS_KEY := $(shell aws secretsmanager get-secret-value --secret-id=ci-user-aws-keys${randomSeed} | jq -r '.SecretString'| jq -r .secret_key)
+	GITHUB_ORG := $(shell echo ${REPOSITORY} | cut -d "/" -f 2)
+	GITHUB_REPO := $(shell echo ${REPOSITORY} | cut -d "/" -f 3)
 	curl -X POST --header "Content-Type: application/json" -d '{"name":"CIRCLECI_API_KEY", "value":"${CIRCLECI_API_KEY}"}' https://circleci.com/api/v1.1/project/github/${GITHUB_ORG}/${GITHUB_REPO}/envvar?circle-token=${CIRCLECI_API_KEY}
 	curl -X POST --header "Content-Type: application/json" -d '{"name":"AWS_ACCESS_KEY_ID", "value":"${AWS_ACCESS_KEY_ID}"}' https://circleci.com/api/v1.1/project/github/${GITHUB_ORG}/${GITHUB_REPO}/envvar?circle-token=${CIRCLECI_API_KEY}
 	curl -X POST --header "Content-Type: application/json" -d '{"name":"AWS_SECRET_ACCESS_KEY", "value":"${AWS_SECRET_ACCESS_KEY}"}' https://circleci.com/api/v1.1/project/github/${GITHUB_ORG}/${GITHUB_REPO}/envvar?circle-token=${CIRCLECI_API_KEY}
 	@echo "\nFollow CIRCLECI project"
 	curl -X POST https://circleci.com/api/v1.1/project/github/${GITHUB_ORG}/${GITHUB_REPO}/follow?circle-token=${CIRCLECI_API_KEY}
 	@echo "\nDone"
 
+
+# The following parameters are required by 'summary':
+# - MODULE_NAME
+# - REPOSITORY
+# - PIPELINE
+# - PRODUCTION_API
+# - STAGING_API
+#
+
+summary:
+	@echo ${MODULE_NAME}
+	@echo "- Repository URL: ${REPOSITORY}"
+	@echo "- Deployment Pipeline URL: ${PIPELINE}"
+	@echo "- Production API: ${PRODUCTION_API}"
```

Comment:
The makefile will be run once per environment which sort of complicates this. If we choose to run both staging and production do we want to collect all this summary output from each run and display it at the end? Maybe for now just show the environment relevant to this execution?

Reply:
Also we should be able to construct the url for the API from the module parameters.

Reply:
I was hoping I could add another function to `internal/apply.apply.go` in `zero` that would be similar to `applyAll` but without the `applyEnvironments` parameter.  It would do a depth first traversal of the module graph and export the module params as env variables, but only run `make summary` for each module once.

Reply:
Okay, that should be fine.

Reply:
For the API URL you should be able to use ${stagingBackendSubdomain}${stagingHostRoot}